roles/KubevirtNodeLabeller/tasks/main.yml

```diff
@@ -4,13 +4,55 @@
     state: present
     definition: "{{ item | from_yaml }}"
   with_items: "{{ lookup('template', 'kubevirt-node-labeller-roles.yaml.j2').split('\n---\n') | select('search', '(^|\n)[^#]') | list }}"
-
+  register: roles
 - name: Create the node labeller daemon set
   k8s:
     state: present
     definition: "{{ lookup('template', 'kubevirt-node-labeller-ds.yaml.j2') | from_yaml }}"
   register: nl
-
+# Actively inject owner references in order to adopt existing resources during an upgrade
```

Comment:
Same here, are you sure you need this? You are reading the objects from file and then posting them. The ansible operator sdk should do the right thing automatically.

Reply:
Again here, if the DS already exists from a previous deployment, the owner reference would not be set

Reply:
What I do not like here is that you list all the different types you want to own. That will make maintenance hard if the source yaml file changes. Can you just iterate over all the created objects instead?

Reply:
Could you take another look please? I've run the role for each resource that needs to be owned by the CR